src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -827,20 +826,16 @@ private SafeCurlSlistHandle SetRequestHeaders(SafeCurlHandle handle, HttpRequest
                 contentHeaders = request.Content.Headers;
             }
 
-            string[] allHeaders = HeaderUtilities.DumpHeaders(request.Headers, contentHeaders)
-                .Split(s_headerDelimiters, StringSplitOptions.RemoveEmptyEntries);
+            // Prepare request headers
+            List<string> allHeaders = PrepareRequestHeaders(request.Headers, contentHeaders);
```

Comment:
Separately, do we need to be concerned with duplicates between request.Headers and contentHeaders?